tools/ports/liblzma.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2020 The Emscripten Authors.  All rights reserved.
+# Emscripten is available under two separate licenses, the MIT license and the
+# University of Illinois/NCSA Open Source License.  Both these licenses can be
+# found in the LICENSE file.
+
+import os
+import shutil
+from tools import building
+
+VERSION = '5.2.5'
+HASH = '7443674247deda2935220fbc4dfc7665e5bb5a260be8ad858c8bd7d7b9f0f868f04ea45e62eb17c0a5e6a2de7c7500ad2d201e2d668c48ca29bd9eea5a73a3ce'
+
+
+def needed(settings):
+  return settings.USE_LIBLZMA
+
+
+def get(ports, settings, shared):
+  ports.fetch_project('liblzma', 'https://tukaani.org/xz/xz-' + VERSION + '.tar.gz', 'xz-' + VERSION, sha512hash=HASH)
+
+  def create():
+    ports.clear_project_build('liblzma')
+
+    source_path = os.path.join(ports.get_dir(), 'liblzma', 'xz-' + VERSION)
+    dest_path = os.path.join(ports.get_build_dir(), 'liblzma')
+
+    shared.try_delete(dest_path)
+    os.makedirs(dest_path)
+    shutil.rmtree(dest_path, ignore_errors=True)
+    shutil.copytree(source_path, dest_path)
+
+    config_args = ['./configure', '--disable-shared']
```

Comment:
It would be good to avoid configure, as that may not work on windows. Is there another build system there, or maybe we can run configure and bundle the output `config.h` or such?

Reply:
I'll check if this works on Windows. I'd rather avoid pulling the guts from build system if possible. It is ugly and harder to maintain.

Reply:
Ok, obviously it won't work on Windows without MinGW shell. There's also cmake available, but if we want building on Windows without other dependencies would miss the point.

I changed it to don't use any build system, by bundling necessary config defines.